static/js/components/NewObservation.jsx

```diff
@@ -34,29 +54,28 @@ const NewObservation = () => {
         title: "Observation file",
         description: "Observation file",
       },
-      instrument_id: {
-        type: "integer",
-        oneOf: instrumentList.map((instrument) => ({
-          enum: [instrument.id],
-          title: `${
-            telescopeList.find(
```

Comment:
And this is where we find the instrument name, which you replaced by a variable called `telescopeName` that doesnt exist at all in your generic component (again, you have to build the option list and then pass it, and not hardcode the label in the generic component that defeats its purpose).